Binaries/DafnyPrelude.bpl

```diff
@@ -952,9 +952,8 @@ function Seq#Append<T>(Seq T, Seq T): Seq T;
 axiom (forall<T> s0: Seq T, s1: Seq T :: { Seq#Length(Seq#Append(s0,s1)) }
   Seq#Length(Seq#Append(s0,s1)) == Seq#Length(s0) + Seq#Length(s1));
 
-// Append preserves $Is
-axiom (forall s0 : Seq Box, s1 : Seq Box, t : Ty :: { $Is(Seq#Append(s0,s1),t) }
-    $Is(s0,t) && $Is(s1,t) ==> $Is(Seq#Append(s0,s1),t));
+axiom (forall s0 : Seq Box, s1 : Seq Box, t : Ty :: { $Is(Seq#Append(s0, s1), TSeq(t)) }
+    $Is(s0, TSeq(t)) && $Is(s1, TSeq(t)) ==> $Is(Seq#Append(s0, s1), TSeq(t)));
```

Comment:
It's not clear that this (corrected) axiom is needed at all, because it follows from the more general axiom

``` boogie
axiom (forall v: Seq Box, t0: Ty :: 
  { $Is(v, TSeq(t0)) } 
  $Is(v, TSeq(t0))
     <==> (forall i: int :: 
      { Seq#Index(v, i) } 
      0 <= i && i < Seq#Length(v) ==> $IsBox(Seq#Index(v, i), t0)));
```

and the definition of `Seq#Append`, which are already in `DafnyPrelude.bpl`.

I did one experiment to see if the more general axiom gets used. The test I wrote is

``` dafny
predicate P(s: seq<int>)

method M(a: seq<int>, b: seq<int>)
  requires forall s: seq<int> :: P(s)
{
  assert P(a + b);
}
```

To prove this `assert`, the verifier has to prove `$Is(Seq#Append(a, b), TSeq(TInt))`. It does prove this property from the general axiom above, with the axiom on L955.

My conclusion is that we don't need this axiom at all and that we should remove it.